### ISSUE TITLE
Items disappear after deselected "In scope" under the History TAB

### ISSUE THREAD

**Issue by zer0uno** [zaproxy/zaproxy#7474]
### Describe the bug

I've noticed that the option "show only URLs in scope" under the History Tab doesn't work properly, when it is enabled and then disabled some requests disappear and can't be found anymore.

### Steps to reproduce the behavior

This is the situation:
- a context with "In scope" flag enabled
- forced user mode enabled

under the history tab no option selected:
![2022-09-26_10-26](https://user-images.githubusercontent.com/15458756/192229158-c442741e-b09c-4aad-9a32-fd7fc67dfc0b.png)

when I manually explore my in scope site then I correctly get all the requests, with the different sources, under the history tab:
![2022-09-26_10-27](https://user-images.githubusercontent.com/15458756/192229538-c9fc7e39-bd67-41e7-8bed-ddc251032d0f.png)

but if I enable "show only URLs in scope" then the requests with "auth" then they disappear (in this case the requests made during the authorization are under the same scope):
![2022-09-26_10-30](https://user-images.githubusercontent.com/15458756/192230232-a34becec-c0fc-4454-9d78-1e80697e3785.png)

and in addition if then I disable "show only URLs in scope" then the requests with "auth" then I cannot find anymore the requests with "auth" source:
![2022-09-26_10-32](https://user-images.githubusercontent.com/15458756/192230493-3786ecb4-4076-4144-85d3-ff38be9b5951.png)

However I've also noticed that if I enable "show only URLs in scope" before manually exploring my in scope site then both the requests with "auth" and "proxy" source are correctly shown. But if then I disable and enable "show only URLs in scope", requests with "auth" source are not shown anymore.

### Expected behavior

When the "show only URLs in scope" option is disabled, after being enabled, all requests must be present.

### Software versions

OWASP ZAP
Version: 2.11.1
---
Installed Add-ons: [[id=alertFilters, version=13.0.0],
[id=ascanrules, version=48.0.0], [id=automation,
version=0.17.0], [id=bruteforce, version=11.0.0],
[id=callhome, version=0.4.0], [id=commonlib,
version=1.10.0], [id=communityScripts, version=14.0.0],
[id=custompayloads, version=0.12.0], [id=diff,
version=11.0.0], [id=directorylistv1, version=5.0.0],
[id=domxss, version=13.0.0], [id=encoder, version=0.6.0],
[id=exim, version=0.2.0], [id=formhandler, version=6.0.0],
[id=fuzz, version=13.7.0], [id=gettingStarted,
version=13.0.0], [id=graaljs, version=0.2.0], [id=graphql,
version=0.10.0], [id=help, version=14.0.0], [id=hud,
version=0.14.0], [id=importurls, version=8.0.0], [id=invoke,
version=11.0.0], [id=network, version=0.2.0], [id=oast,
version=0.11.0], [id=onlineMenu, version=9.0.0],
[id=openapi, version=28.0.0], [id=pscanrules,
version=43.0.0], [id=pscanrulesAlpha, version=36.0.0],
[id=pscanrulesBeta, version=30.0.0], [id=quickstart,
version=34.0.0], [id=replacer, version=10.0.0], [id=reports,
version=0.15.0], [id=retest, version=0.3.0], [id=retire,
version=0.15.0], [id=reveal, version=4.0.0],
[id=saverawmessage, version=6.0.0], [id=savexmlmessage,
version=0.2.0], [id=scripts, version=31.0.0], [id=selenium,
version=15.10.0], [id=soap, version=14.0.0], [id=spiderAjax,
version=23.9.0], [id=tips, version=9.0.0],
[id=webdriverlinux, version=43.0.0], [id=websocket,
version=26.0.0], [id=zest, version=36.0.0]]
---
Operating System: Linux
Java Version: Ubuntu 11.0.16
System's Locale: en_US
Display Locale: en_GB
Format Locale: en_US
ZAP Home Directory: ***************************************
ZAP Installation Directory: /opt/zaproxy/./
Look and Feel: FlatLaf Darcula (com.formdev.flatlaf.FlatDarculaLaf)

### Would you like to help fix this issue?

- [ ] Yes

**Comment by thc202** [comment]
That's the expected behaviour the auth requests are temporary.

**Comment by zer0uno** [comment]
it's not very intuitive and if I click by mistake on "show only URLs in Scope" then I'm lost forever, there is no possibility to go back and I have to repeat everything from scratch

**Comment by thc202** [comment]
Those are removed either way, if you close the session and open it again they will no longer be there. You can authenticate again if you want to see authentication requests.

**Comment by zer0uno** [comment]
Yes, but conceptually speaking, what request history tab is it if it doesn't keep the history? And what's to advantage to have "auth" request disappearing?

**Comment by zer0uno** [comment]
It would be useful and more consistent to have a button as the following:
![2022-09-26_12-50](https://user-images.githubusercontent.com/15458756/192258739-8bbcf8e1-b766-4b53-9fca-df9ca993d2da.png)

The button is present in many tabs, why not having it in **all** the tabs? It would make the environment more consistent and user-friendly
